### PR TITLE
[MRI_Violations] Fixes permission bug; integration test to confirm bug fix; test plan update

### LINKED ISSUE #5419: [MRI Violations] No Series UID column
There is a filter in the form called Series UID but there is no corresponding column.

Entering data in the filter still causes columns to be hidden.

Not sure if this is a code issue or something to do with Raisinbread.

Testing on `test-dev-loris.ca`.

### LINKED ISSUE #5421: [MRI Violations] No column named "TarchiveID"
TestPlan:

> MRI Protocol Check Violations page - check Selection Filter results for each of Tarchive ID, PatientName, CandID and DICOM Series UID [Manual Testing]

There is a filter called TarchiveID but it's not clear what column that corresponds to. 

Entering data in the filter causes some rows to disappear so it appears that it's operating on something.

Not sure if this is a UI bug or a DB issue.

### LINKED ISSUE #5422: [MRI Violations] Request for help with imaging testing tasks
> 8. Verify one or two of the scans in the second table and ensure it indeed violates the MRI protocol. [Manual Testing]

> 12. Upload scan that violates the mri_protocol table and ensure it shows up in the module as 'Could not identify scan type' [Manual Testing]

> 13. Upload scan that violates the mri_protocol_checks table and ensure it shows up in the module as Protocol Violation [Manual Testing]

I am not familiar enough with imaging to test these points.  Could someone help with this?

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

Allows users with the 'edit' permission to view the module even if they don't have 'view' permissions.

Adds an integration test to the above + code cleanup.

Rewrites the test plan to be up-to-date and easier to follow.

#### Links to related tickets (GitHub, Redmine, ...)

* Resolves #5419 
* Resolves #5421 
* Resolves #5422

### CODE REVIEW

File: modules/mri_violations/test/TestPlan.md [hunk 1]
```diff
@@ -1,69 +1,46 @@
...
+3. The `Problem` column with the entry `Could not identify scan type` should 
+   link to the `MRI Protocol Violations` page.
+4. Resolve issues with each of the different types of resolutions, and ensure 
+    that they show up in the `Resolved` tab, and are no longer in the 
+    `Not resolved` tab.
+5. Ensure that the link to the protocol violation works within the `Resolved` 
+    tab.
+6. Ensure all filters and sorting work in the `Resolved` tab.
+
+### MRI Protocal Violations Page
```
Comment (cmadjar) [inline modules/mri_violations/test/TestPlan.md]: Typo in Protocol

**Comment by johnsaigle** [comment]
@cmadjar fixed!

**Comment by driusan** [comment]
@johnsaigle travis

**Comment by johnsaigle** [comment]
Blocked by #5561 which adds the ability for users to view the module with only the edit permission.

**Comment by johnsaigle** [comment]
This PR has absorbed #5561 because that's easier for everybody

File: modules/mri_violations/php/mri_protocol_violations.class.inc [hunk 2]
```diff
@@ -20,7 +20,7 @@ namespace LORIS\mri_violations;
  *
  * @category MRI
  * @package  Main
- * @author   Zia Mohadesz <zia.mohades@gmail.com>
+ * @author   Zia Mohades <zia.mohades@gmail.com>
```
Comment (driusan) [inline modules/mri_violations/php/mri_protocol_violations.class.inc]: I don't think you should change other people's names..

Comment (johnsaigle) [reply modules/mri_violations/php/mri_protocol_violations.class.inc]: It's a typo though?

Comment (driusan) [reply modules/mri_violations/php/mri_protocol_violations.class.inc]: @mohadesz sometimes spells his name with a z.

Comment (johnsaigle) [reply modules/mri_violations/php/mri_protocol_violations.class.inc]: Ah i thought the username was mohadesz[ia].

Comment (johnsaigle) [reply modules/mri_violations/php/mri_protocol_violations.class.inc]: anyway I don't really care about this

Comment (johnsaigle) [inline modules/mri_violations/php/mri_protocol_violations.class.inc]:
```suggestion
 * @author   Zia Mohadesz <zia.mohades@gmail.com>
```

**Pull request merged**